Wikipedia/Code/ExploreViewController.swift

```diff
@@ -230,6 +227,24 @@ class ExploreViewController: UIViewController, WMFExploreCollectionViewControlle
         showSearchBar(animated: false)
     }
     
+    var statusBarUnderlay: UIView?
+    
+    func exploreCollectionViewController(_ collectionVC: WMFExploreCollectionViewController, willEndDragging scrollView: UIScrollView, velocity: CGPoint) {
+        self.navigationController?.setNavigationBarHidden(velocity.y > 0, animated: true)
+    }
+    
+    func addStatusBarUnderlay() {
+        let statusBarUnderlay = UIView()
+        statusBarUnderlay.translatesAutoresizingMaskIntoConstraints = false
+        let topConstraint = self.view.topAnchor.constraint(equalTo: statusBarUnderlay.topAnchor)
```

Comment:
According to the docs the subview should call `constraint(equalTo`. Same bug in the web vc's `addHeaderView`.

Reply:
The `subview` in the docs is just an example. In practice, for constraints that are just `equalTo` it doesn't matter which item is first. Think of it like an equation `view.topAnchor = subview.topAnchor` is equivalent to `subview.topAnchor = view.anchor`.

Reply:
I thought it made a difference though - you want to set the new item equal to something that already exists... that's why even in storyboard it has option for reversing first and second item...

Reply:
It can make a difference if it's not just an equality constraint. In this case, it's fine.

Reply:
This https://stackoverflow.com/a/28164113/135557  cleared it up for me... order makes a difference but only when it evaluates the constant. 

This reply also helped:

The formula is an equation, not an assignment. It's not "set firstItem.firstItemAttribute to be the result of …". It's "keep this equation true" or "keep firstItem.firstItemAttribute equal to …". That is symmetric. The system tries to find values for all of the attributes of all of the items that satisfies all of the constraints. Logically, it doesn't start with some values and work from those to compute the others.